services/webos-service-remote/sshbus.js

```diff
@@ -0,0 +1,116 @@
+import EventEmitter from 'events';
+import { readFileSync } from 'fs';
+import Message from './message';
+import { SSH } from './ssh-promise';
+
+class Request extends EventEmitter {
+    constructor(ssh) {
+        super();
+        this.ssh = ssh;
+    }
+
+    cancel() {
+        this.ssh.close();
+    }
+}
+
+export default class Handle {
+    constructor(service) {
+        this.service = service;
+    }
+    /**
+     * Send a message via luna-send-pub and start interactive session
+     * @param {string} uri 
+     * @param {string} payload 
+     * @returns EventEmitter of this session
+     */
+    subscribe(uri, payload) {
+        const ssh = new SSH();
+        var request = new Request(ssh);
+        this._getSshConfig().then(config => ssh.connect(config))
+            .then(() => ssh.spawn('luna-send-pub', ['-i', uri, payload]))
+            .then(stream => {
+                var stdout = '';
+                stream.on('close', (code, signal) => {
+
+                });
+                stream.on('data', data => {
+                    stdout += data.toString();
+                    var searchPos = 0, breakPos = 0;
+                    while ((breakPos = stdout.indexOf('\n', searchPos)) > 0) {
+                        const line = stdout.substring(searchPos, breakPos).trim();
+                        request.emit('response', Message.constructBody(line, true));
+                        searchPos = breakPos + 1;
+                    }
+                    if (searchPos) {
+                        stdout = stdout.substring(searchPos);
+                    }
+                });
+            })
+            .catch(err => request.emit('cancel', Message.constructBody(JSON.stringify({
+                returnValue: false,
+                errorCode: -1,
+                errorText: `Unable to exec luna-send-pub: ${err}`,
+            }), false)))
+            .catch(reason => {
+                if (reason.stack) {
+                    console.error(reason.stack);
+                }
+                request.emit('cancel', Message.constructBody(JSON.stringify({
+                    returnValue: false,
+                    errorCode: -1,
+                    errorText: `Failed to start ssh session: ${reason}`
+                }), false));
+            });
+        return request;
+    }
+    /**
+     * Send a message via luna-send-pub and get response
+     * @param {string} uri 
+     * @param {string} payload 
+     * @returns Promise to response message
+     */
+    async call(uri, payload) {
+        const ssh = new SSH();
+        return this._getSshConfig().then(config => ssh.connect(config)
```

Comment:
you could use async/await syntax here, and I feel this would be a little bit more readable:
```js
const config = await this._getSshConfig();
await ssh.connect(config);

const { stdout } = await ssh.exec('luna-send-pub', ...)
return Message.constructBody(stdout.trim(), false);
```

Reply:
I agree, this would be better. Another n00b question, would it be safe to change for `subscribe` as well? I'm not sure if callers of this class can enable `async` though...

Reply:
I don't need it for `subscribe` yet, so I will only use `async` for `call` now.